compiler/ztests/quoted-named-type.yaml

```diff
@@ -0,0 +1,8 @@
+zed: |
+  yield <"@foo"={x:int64}>
+
+input: |
+  null
+
+output: |
+  <"@foo"={x:int64}>
```

Comment:
Nit: This is primarily a parser test so maybe put it in compiler/parser/ztests?

Reply:
Sure